From 9d88c45740d48d4fd5c77c6012731189c2acb6e6 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 24 Dec 2014 09:53:49 +0100
Subject: [PATCH] Try to run freerouting locally

The web site is no longer available and freerouting turned free software.
Fedora ships it.
---
 pcbnew/dialogs/dialog_freeroute_exchange.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/pcbnew/dialogs/dialog_freeroute_exchange.cpp b/pcbnew/dialogs/dialog_freeroute_exchange.cpp
index ca3bef0..2dca971 100644
--- a/pcbnew/dialogs/dialog_freeroute_exchange.cpp
+++ b/pcbnew/dialogs/dialog_freeroute_exchange.cpp
@@ -126,7 +126,15 @@ void DIALOG_FREEROUTE::OnLaunchButtonClick( wxCommandEvent& event )
 {
     wxString url;
     wxString command;
-    wxFileName fileName( FindKicadFile( wxT( "freeroute.jnlp" ) ), wxPATH_UNIX );
+    wxFileName fileName( FindKicadFile( wxT( "freerouting" ) ), wxPATH_UNIX );
+
+    if( fileName.FileExists() )
+    {
+        ProcessExecute( fileName.GetFullPath() );
+        return;
+    } else {
+        fileName.Assign ( FindKicadFile( wxT( "freeroute.jnlp" ) ), wxPATH_UNIX );
+    }
 
     if( fileName.FileExists() )
     {
-- 
2.1.0

